File: src/frontend/src/app/components/atendimento/agendamento/agendaregra/model/agenda-regra-grid-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class AgendaRegraGridDTO {
  @Expose() id?: string;
  @Expose() agendaId?: string;
  @Expose() dataInicio?: string;
  @Expose() dataFim?: string;
  @Expose() horaInicio?: string;
  @Expose() horaFim?: string;
  @Expose() duracaoSessaoMinutos?: number;
  @Expose() diasSemanaFormatado?: string;
  @Expose() qtdConvenios?: number;
  @Expose() qtdProcedimentos?: number;
  @Expose() deleted?: boolean;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/agendamento/agendaregra/dto/AgendaRegraGridDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.agendamento.agendaregra.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonFormat;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AgendaRegraGridDTO implements DTO, GridDTO {

    private UUID id;
    private UUID agendaId;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dataInicio;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dataFim;

    @JsonFormat(pattern = "HH:mm")
    private LocalTime horaInicio;

    @JsonFormat(pattern = "HH:mm")
    private LocalTime horaFim;

    private Integer duracaoSessaoMinutos;
    private String diasSemanaFormatado;
    private Integer qtdConvenios;
    private Integer qtdProcedimentos;

    private LocalDateTime createdAt;
    private Boolean deleted;
}
